mitodl/ocw_oer_export
# Use OCW FM exported keywords for CR_KEYWORDS

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/3378

### Description (What does it do?)
Uses FM exported keywords for `CR_keywords` field in OER template. Previously we were just using OCW course topics as keywords. 
Now we are relying on [FM exported keywords](https://github.com/mitodl/hq/issues/2426#issuecomment-1883612084). If they're not available for a course, then we use course's topics as keywords.

We were able to get 2269/2496 OCW courses in FM list. 


### Updated CSV
Here is the updated CSV file (3.1mb):
https://docs.google.com/spreadsheets/d/1zT0atM14ZbBx7fcy1NWfr94pdhygbIJgKgKaM661krM/edit#gid=1443282546

### How to test
1. Look at the updated CSV file above. Observe `CR_KEYWORDS`
2. Make sure there isn't anything weird or inconsistent in that column. By that I mean, in the file, `ocw_oer_export/mapping_files/fm_keywords_export.csv`, I found different cases:
    - Comma separated keywords
    - Semicolon separated keywords
    - Newline separated, etc (and multiples of those). Make sure there isn't anything else.

Now checkout to this branch, run the program, make sure everything is good:
3. Build the container:

```
docker compose build
```
4. Start the container:

```
docker compose run --rm app
```

5. To generate a JSON file containing complete API data:

```
docker compose run --rm app --create_json
```

6. To create a CSV file from the local JSON file:

```
docker compose run --rm app --create_csv --source=json
```

7. To run unit tests:

```
docker run --rm ocw_oer_export python -m unittest discover
```
